eth_account/account.py

```diff
@@ -50,7 +57,30 @@
 )
 
 
+class _AccountModule(ModuleType):
+
+    '''
+    TODO - Documentation
+    '''
+    _defaultKDF = 'scrypt'
+
+    @property
+    def defaultKDF(self):
+        '''
+        TODO - Documentation
+        '''
+        return type(self)._defaultKDF
+
+    @defaultKDF.setter
+    def defaultKDF(self, kdf):
+        type(self)._defaultKDF = kdf
+
+
+sys.modules[__name__].__class__ = _AccountModule
```

Comment:
I can switch it. I initally wanted to put it on one of the Account classes, but the issue statement:
>Allow setting the default KDF using both environment variables and via a property on the web3.eth.account module.
>
> `web3.eth.account.defaultKDF = 'scrypt'`

Made it seem like you wanted it on `eth_account.account` (the module) and not `eth_account.account.Account` (the class).

Reply:
You correctly interpreted my original comment/spec (which I didn't recall until you just pointed it out), but seeing it in action, I'm thinking that class level default with option for global default via environment variable is the right way to go.

Reply:
Note that in the web3 context, `w3.eth.account` is an instance of the `Account` class, so changing the default works smoothly as `w3.eth.account.defaultKDF = 'scrypt'`.

Reply:
So I'm still a little unclear about the ask here...

I have it setup so it's a class variable that gets instantiated with `os.getenv('ETH_ACCOUNT_KDF', 'scrypt')`. To set the env var after the class had already been imported, I had to reload the module and use the new Account reference to verify this functionality ([see here](https://github.com/ethereum/eth-account/pull/38/commits/8aec325d1b02c045143b0106e6a807524c04c6b6#diff-1de692d823115945c2232e1aa3066811R109)). Based on the docs [here](https://docs.python.org/3/library/importlib.html#importlib.reload):
>Other references to the old objects (such as names external to the module) are not rebound to refer to the new objects and must be updated in each namespace where they occur if that is desired.

I think this is the proper way to do it as the
```
from eth_account import (
    Account,
)
```
at the top of the file would be the original reference, where the class object has `default_kdf = 'scrypt'`. Still kind of new to Python, so please let me know if my understanding of this is off.

Another question I had @carver, based on your comment, is if we want this value to be settable vs. something that's just instantiated with a default and that default can be overridden by just providing a KDF to the `encrypt` method. I'm thinking that making it mutable just adds more complexity since the only place it's used is the `encrypt` method and that already allows you to provide whatever KDF you want.

Commit: https://github.com/ethereum/eth-account/pull/38/commits/8aec325d1b02c045143b0106e6a807524c04c6b6#diff-a9c6ca5d3c0ddc6c626bcee1142503c9